### ISSUE TITLE
Table export function in Dbeaver has a bug

### ISSUE THREAD

**Issue by gbaremz** [dbeaver/dbeaver#22442]
### Description

My Dbeaver Version is "Version 23.3.1.202312241705" and make use of PostgreSQL DB.
Ever since I upgraded my previous Dbeaver community version I have noticed a consistent bug. 
For example, imagine a table name in Database-A is named "version_A22",  if I export this table into another Database-B, using the Dbeaver export option that comes up by right click on mouse, the new name in Database-B would then be "version_a22". 
Apparently Dbeaver needs to include "spcialtablename " in its table names duing table selection. 

### DBeaver Version

23.3.1.202312241705

### Operating System

Ubuntu 20.04.6 LTS

### Database and driver

PostgreSQL JDBC Driver

### Steps to reproduce

1. Right click on a table
2. select export data 
3. select (Export Data) and click next
4. select (Tables mapping) choose necessary destination database name and click next....
5.  select (Extraction settings) and click next
6. (Data load settings) and click next
7. (Confirm) and click proceed
8. After export ends, check table name and the error in "Description" above would happen.

**Comment by MashaKorax** [comment]
Thanks for your report. It happens after table creation (not only export)


https://github.com/dbeaver/dbeaver/assets/84867187/dd631935-0ac2-45eb-97f1-d495b322c2f2

**Comment by LonwoLonwo** [comment]
Hello @gbaremz 

Thanks for the bug report.

![2024-01-12 15_30_14-Configure metadata structure](https://github.com/dbeaver/dbeaver/assets/45152336/71615c35-cb9d-439d-9e5c-55345baee693)

Probably, we need to keep quoted names in quotes during data transfer inside one database.

As a workaround for now - you can add quotes and change the table name in the data transfer settings

![2024-01-12 15_34_34-Configure metadata structure](https://github.com/dbeaver/dbeaver/assets/45152336/cee3290c-233a-416e-89a1-dd072122058a)

**Comment by gbaremz** [comment]
Ja, but it's just a longshot compared to the ease one had before the Upgrade. I just get the feeling that each upgrade results to something good and something quite dangerous for Data guys.
